designs/NuGet-RemoteFileService.md

```diff
@@ -0,0 +1,70 @@
+
+# Spec Name
+
+* Status: In Review
+* Author(s): [Rob Relyea](https://github.com/rrelyea)
+* Issue: [10151](https://github.com/NuGet/Home/issues/10151) INuGetRemoteFileService - Fetch Images and embedded licenses for codespaces and normal execution
+
+## Problem Background
+
+With Codespaces support added to NuGet client, our code needs to be able to handle all packages and communication with feeds happening on a server node. This is architected as async method calls to services that are remoted. In most cases, we also have chosen to have 1 code path, for normal execution and remoted execution. These services are still used in the normal execution, however, the service runs in the same process, as opposed to being remoted across the network.
+
+Package Icons can either come from:
+- a package feeds search service
+- (old style) or a URI to an arbitrary place on the web
+- embedded in a package
+
+Licenses can also come from a few places, including a license file embedded in a package.
+
+## Who are the customers
+
+All PM UI customers.
+
+## Goals
+
+Support remote and normal fetching of icons and licenses.

```

Comment:
Again, use of the word "normal" is confusing.

From a client perspective, how the client fetches package icons and license metadata should be consistent, whether the client is operating in standalone or Codespaces-connected mode.

From a service perspective --- whether the service is running locally or remotely ---, the service should support fetching package icons and license metadata from server-local data sources (e.g.:  global packages folder) or from server-remote data sources (like a remote package source).

Reply:
followed naming from other suggestion: standalone and Codespaces-connected. see next iteration.